monitoring/linear_combination.py

```diff
@@ -0,0 +1,100 @@
+from operator import itemgetter
+
+import iris
+from scriptengine.exceptions import (
+    ScriptEngineTaskArgumentInvalidError,
+    ScriptEngineTaskArgumentMissingError,
+    ScriptEngineTaskRunError,
+)
+from scriptengine.tasks.core import Task, timed_runner
+
+import helpers.cubes
+
+
+class LinearCombination(Task):
+
+    _required_arguments = (
+        "src",
+        "dst",
+    )
+
+    def __init__(self, arguments):
+        LinearCombination.check_arguments(arguments)
+        super().__init__(arguments)
+
+    @timed_runner
+    def run(self, context):
+        self.log_info("Create new netCDF file from linear combination of inputs")
+
+        sources = self.getarg("src", context)
+        if not isinstance(sources, list):
+            self.log_error("Invalid 'src' argument, must be a list")
+            raise ScriptEngineTaskArgumentInvalidError
+
+        result = 0  # Initialize sum of scaled cubes
+        for src in sources:
+            self.log_debug(f"Processing source {src}")
+            try:
+                path, varname = itemgetter("path", "varname")(src)
+            except TypeError:
+                self.log_error("The list items of the 'src' argument must be dicts")
+                raise ScriptEngineTaskArgumentInvalidError
+            except KeyError as e:
+                self.log_error(f"Missing key(s) {e} in 'src' list item")
+                raise ScriptEngineTaskArgumentMissingError
+
+            try:
+                factor = float(src.get("factor", 1.0))
+            except (ValueError, TypeError) as e:
+                self.log_error(
+                    f"Unable to convert the 'factor' argument to a number: {e}"
+                )
+                raise ScriptEngineTaskArgumentInvalidError
+
+            summand = helpers.cubes.load_input_cube(path, varname)
+            summand = helpers.cubes.remove_aux_time(summand)
+
+            try:
+                scaled_summand = factor * summand
+            except Exception as e:
+                self.log_error(f"Error scaling '{summand.name()}': {e}")
+                raise ScriptEngineTaskRunError
+            try:
+                scaled_summand.convert_units(summand.units)
+            except iris.exceptions.UnitConversionError as e:
+                self.log_warning(f"Unit conversion issue: {e}")
```

Comment:
Again, the question here would be if this exception actually gets thrown in realistic scenarios. This would mean that Iris implicitly converted the unit of summand while scaling but is not able to convert it back?

Reply:
Same argument here: I thought it was complex enough to justify the extra logging, but maybe it's overkill.

The simpler, the better - I'll remove the exception handling in both cases.